src/Camera/QGCCameraControl.cc

```diff
@@ -1171,12 +1171,14 @@ void
 QGCCameraControl::_requestAllParameters()
 {
     //-- Reset receive list
-    for(const QString& paramName: _paramIO.keys()) {
-        if(_paramIO[paramName]) {
-            _paramIO[paramName]->setParamRequest();
+    auto iterator = _paramIO.constBegin();
+    while (iterator != _paramIO.constEnd()) {
+        if (iterator.value()) {
+            iterator.value()->setParamRequest();
         } else {
-            qCritical() << "QGCParamIO is NULL" << paramName;
+            qCritical() << "QGCParamIO is NULL" << iterator.key();
         }
+        ++iterator;
```

Comment:
Nak. even tougth iterator is faster than creating a array of keys, this version is worse to read and easier to create mistakes. sometimes it's more iportant readability than speed, and looking at what the code is trying to do, the old, and the new versions are wrong.

for (const auto param : qAsConst(_paramIO) {
 if (param) {
   param->setParamRequest();
 } 
}

if we are only using the name of the param to print on the qCritcal(), the param should have a ->name() method.